zerver/lib/import_realm.py

```diff
@@ -689,6 +689,46 @@ def bulk_import_model(data: TableData, model: Any, dump_file_id: Optional[str] =
         logging.info("Successfully imported %s from %s[%s].", model, table, dump_file_id)
 
 
+def fix_user_groups_data(data: TableData) -> None:
+    # Old exports might not contain the "zerver_namedusergroup" data.
+    data["zerver_namedusergroup"] = []
+    for group in data["zerver_usergroup"]:
+        data["zerver_namedusergroup"].append(
+            {
+                "usergroup_ptr_id": group["id"],
+                "realm_for_sharding_id": group["realm_id"],
+                "named_group_name": group["name"],
+                "named_group_description": group["description"],
+                "named_group_is_system_group": group["is_system_group"],
+                "named_group_can_mention_group_id": group["can_mention_group_id"],
+            }
+        )
+
+
+def bulk_import_named_user_groups(data: TableData) -> None:
+    vals = [
+        (
+            group["usergroup_ptr_id"],
+            group["realm_for_sharding_id"],
+            group["named_group_name"],
+            group["named_group_description"],
+            group["named_group_is_system_group"],
+            group["named_group_can_mention_group_id"],
+        )
+        for group in data["zerver_namedusergroup"]
+    ]
+
+    query = SQL(
+        """
+        INSERT INTO zerver_namedusergroup (usergroup_ptr_id, realm_id, name, description, is_system_group, can_mention_group_id)
+        VALUES %s
+        ON CONFLICT DO NOTHING
+        """
+    )
+    with connection.cursor() as cursor:
```

Comment:
Here also, `UserGroup` and `NamedUserGroup` objects are created in separate queries.